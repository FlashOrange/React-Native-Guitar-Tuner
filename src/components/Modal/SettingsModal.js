import React from 'react';
import {
  Text,
  View,
  TouchableOpacity,
  StyleSheet,
  ScrollView,
} from 'react-native';
import Overlay from 'react-native-modal-overlay';
import { colors } from '../../utils';
import ModalButton from './ModalButton';
import instrumentTypes from '../../utils/instrumentTypes';

const SettingsModal = props => {
  const selectButtons = Object.keys(instrumentTypes).map(instrumentKey => {
    return (
      <ModalButton
        text={instrumentTypes[instrumentKey].displayName}
        key={instrumentKey}
        style={
          instrumentKey === props.selectedInstrument
            ? styles.selectedInstrument
            : styles.text
        }
        pressed={() => props.onSelectInstrument(instrumentKey)}
      />
    );
  });
  return (
    <Overlay
      visible={props.visible}
      animationType="zoomInDown"
      onClose={props.onClose}
      closeOnTouchOutside
      containerStyle={{ backgroundColor: colors.overlay }}
      childrenWrapperStyle={styles.childrenWrapper}
      animationDuration={500}
    >
      <Text style={styles.text}>Select An Instrument:</Text>
      <ScrollView>{selectButtons}</ScrollView>
    </Overlay>
  );
};

const styles = StyleSheet.create({
  text: {
    fontSize: 22,
    padding: 8,
  },
  selectedInstrument: {
    fontSize: 24,
    padding: 8,
    fontWeight: 'bold',
  },
  childrenWrapper: {
    backgroundColor: '#eee',
    alignItems: 'stretch',
    borderRadius: 4,
  },
});

export default SettingsModal;
