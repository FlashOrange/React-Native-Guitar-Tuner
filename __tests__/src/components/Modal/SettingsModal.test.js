import React from 'react';
import 'react-native';
import SettingsModal from '../../../../src/components/Modal/SettingsModal';
import renderer from 'react-test-renderer';

test('it renders correctly', () => {
  const tree = renderer.create(<SettingsModal />).toJSON();
  expect(tree).toMatchSnapshot();
});
