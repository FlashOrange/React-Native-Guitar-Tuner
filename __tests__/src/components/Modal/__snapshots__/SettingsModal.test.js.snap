// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<Modal
  animationOutType="fadeOut"
  animationType="none"
  hardwareAccelerated={false}
  onClose={[Function]}
  onRequestClose={[Function]}
  transparent={true}
  visible={false}
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.68)",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 0,
        "padding": 20,
      }
    }
    testID={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "padding": 20,
          },
          Object {
            "alignItems": "stretch",
            "backgroundColor": "#eee",
            "borderRadius": 4,
          },
        ]
      }
      testID={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 22,
            "padding": 8,
          }
        }
      >
        Select An Instrument:
      </Text>
      <RCTScrollView>
        <View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 22,
                  "padding": 8,
                }
              }
            >
              Guitar
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 22,
                  "padding": 8,
                }
              }
            >
              Bass (4-String)
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 22,
                  "padding": 8,
                }
              }
            >
              Bass (5-String, Low-B)
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 22,
                  "padding": 8,
                }
              }
            >
              Bass (5-String, High-C)
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 22,
                  "padding": 8,
                }
              }
            >
              Ukulele
            </Text>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</Modal>
`;
